sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -1040,17 +1034,11 @@ class Column(val expr: Expression) extends Logging {
    *   df.select($"colA".name("colB"))
    * }}}
    *
-   * If the current column has metadata associated with it, this metadata will be propagated
-   * to the new column.  If this not desired, use `as` with explicitly empty metadata.
-   *
    * @group expr_ops
    * @since 2.0.0
    */
   def name(alias: String): Column = withExpr {
-    normalizedExpr() match {
-      case ne: NamedExpression => Alias(expr, alias)(explicitMetadata = Some(ne.metadata))
```

Comment:
This was added by https://github.com/apache/spark/pull/8215. But looking at the added tests at that time, we just want to carry over the metadata from the child of `Alias`, which doesn't need this change as `Alias.metedata` can inherit its child's metadata.